admin_panel/exams/serializer.py

```diff
@@ -0,0 +1,69 @@
+from rest_framework import serializers
+from admin_panel.exams.model import Exam
+from admin_panel.organizations.serializer import OrganizationSerializer
+
+
+class ExamCreateSerializer(serializers.ModelSerializer):
+    """Serializer for creating an exam"""
+    class Meta:
+        model = Exam
+        fields = [
+            'name',
+            'objective',
+            'duration',
+            'difficulty',
+            'department',
+            'semester',
+            'subject',
+            'exam_code',
+            'total_questions',
+            'total_marks',
+            'organization'
+        ]
+
+    def validate(self, data):
+        if data.get('total_marks', 0) < data.get('total_questions', 0):
+            raise serializers.ValidationError(
+                "Total marks cannot be less than total questions"
+            )
+        return data
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider consolidating duplicate validation logic</b></div>
<div id="fix">

Consider consolidating the validation logic for `total_marks` and `total_questions` into a single method to avoid code duplication. This could be achieved by creating a separate method that both `ExamCreateSerializer` and `ExamSerializer` can call.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -1,6 +1,13 @@
  from rest_framework import serializers
  from admin_panel.exams.model import Exam
  from admin_panel.organizations.serializer import OrganizationSerializer
 
 +def validate_total_marks_and_questions(data):
 +    if data.get('total_marks', 0) < data.get('total_questions', 0):
 +        raise serializers.ValidationError(
 +            "Total marks cannot be less than total questions"
 +        )
 +    return data
 +
  class ExamCreateSerializer(serializers.ModelSerializer):
      """Serializer for creating an exam"""
      class Meta:
 @@ -22,11 +29,7 @@
          ]
 
      def validate(self, data):
 -        if data.get('total_marks', 0) < data.get('total_questions', 0):
 -            raise serializers.ValidationError(
 -                "Total marks cannot be less than total questions"
 -            )
 -        return data
 +        return validate_total_marks_and_questions(data)
 
 
  class ExamSerializer(serializers.ModelSerializer):
 @@ -36,12 +39,7 @@
          fields = "__all__"
 
      def validate(self, data):
 -        if all(k in data for k in ['total_marks', 'total_questions']):
 -            if data['total_marks'] < data['total_questions']:
 -                raise serializers.ValidationError(
 -                    "Total marks cannot be less than total questions"
 -                )
 -        return data
 +        return validate_total_marks_and_questions(data)
 
 
  class ExamResponseSerializer(serializers.ModelSerializer):
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/sleep-404/ExamEvoke/pull/8#issuecomment-2499580485>#1f7607</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged